src/app/hooks/useClickTracker/index.jsx

```diff
@@ -1,96 +1,81 @@
 /* eslint-disable no-console */
-import { useEffect, useRef, useContext, useCallback } from 'react';
+import { useContext, useCallback, useState } from 'react';
+import path from 'ramda/src/path';
+import pathOr from 'ramda/src/pathOr';
+
 import { sendEventBeacon } from '#containers/ATIAnalytics/beacon/index';
-import { ServiceContext } from '#contexts/ServiceContext';
-import { RequestContext } from '#contexts/RequestContext';
-import { buildATIClickParams } from '#containers/ATIAnalytics/params';
 import { isValidClick } from './clickTypes';
+import { EventTrackingContext } from '#app/contexts/EventTrackingContext';
 
 const EVENT_TYPE = 'click';
 
-const useClickTracker = ({
-  pageData,
-  componentName,
-  campaignName,
-  format = '',
-  href,
-} = {}) => {
-  let pageIdentifier;
-  let platform;
-  let statsDestination;
-
-  const requestContext = useContext(RequestContext);
-  const serviceContext = useContext(ServiceContext);
-  const { service } = serviceContext;
+const useClickTrackingHandler = (props = {}) => {
+  const campaignName = path(['campaignName'], props);
+  const componentName = path(['componentName'], props);
+  const href = path(['href'], props);
+  const format = pathOr('', ['format'], props);
 
-  const clickRef = useRef(null);
+  const [clicked, setClicked] = useState(false);
+  const { pageIdentifier, platform, service, statsDestination } = useContext(
+    EventTrackingContext,
+  );
 
   const changeUserLocationIfDefined = useCallback(url => {
     if (url) {
       window.location.assign(url);
     }
   }, []);
 
-  try {
-    ({ pageIdentifier, platform, statsDestination } = buildATIClickParams(
-      pageData,
-      requestContext,
-      serviceContext,
-    ));
-  } catch (error) {
-    console.error(
-      `ATI Event Tracking Error: Could not parse tracking values from page data:\n${error.message}`,
-    );
-  }
-
-  const handleClick = useCallback(
+  return useCallback(
     event => {
-      event.stopPropagation();
-      event.preventDefault();
-
-      if (isValidClick(event)) {
-        const nextPageUrl = href || event.target.href;
-        clickRef.current?.removeEventListener('click', handleClick);
+      if (!clicked) {
+        event.stopPropagation();
+        event.preventDefault();
```

Comment:
if we moved these lines inside of the if statement on line 35 then we wouldn't need an else which means we don't need to define the function `changeUserLocationIfDefined`

Reply:
Good point, but I think moving them to inside the line 39 if statement is the one that enables us to remove the else, since the else is in case the event shouldn't be sent